dvc/repo/get.py

```diff
@@ -55,29 +59,20 @@ def get(url, path, out=None, rev=None):
                 # Also, we can't use theoretical "move" link type here, because
                 # the same cache file might be used a few times in a directory.
                 repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
-
-                try:
-                    output = repo.find_out_by_relpath(path)
-                except OutputNotFoundError:
-                    output = None
-
-                if output and output.use_cache:
+                output = repo.find_out_by_relpath(path)
+                if output.use_cache:
                     _get_cached(repo, output, out)
                     return
-
-                # Either an uncached out with absolute path or a user error
-
-                repo_dir = repo.root_dir
-
-        except NotDvcRepoError:
-            # Not a DVC repository, continue below and copy from git
+                # Non-cached output, fall through and try to copy from git.
+        except (NotDvcRepoError, NoOutputInExternalRepoError):
+            # Not a DVC repository or, possibly, path is not tracked by DVC.
+            # Fall through and try to copy from git.
             pass
 
         if os.path.isabs(path):
             raise FileNotFoundError
 
-        if not repo_dir:
-            repo_dir = cached_clone(url, rev=rev)
```

Comment:
We don't reuse repo_dir anymore from external_repo, but:
1) Using repo_dir outside of external_repo context is dangerous, as it is out of its context. E.g. imagine if we decide to delete it at some point. 
2) It wasn't really necessary as we cache clones anyways and copying a cache is fast; And without reusing code is much more compact.